Public calender link broken

### Steps to reproduce

1. Share a calender with a public link
2. Access the public link via a browser
3. 


### Expected behavior

Public calender view should appear

### Actual behaviour

No entrys, the calender list stuck in loading animation
![Screenshot 2022-03-09 204506](https://user-images.githubusercontent.com/88290076/157521590-1c38842b-646e-41c1-96b5-9bed6ff37d23.png)



### Calendar app version

3.2.0

### CalDAV-clients used

_No response_

### Browser

Firefox 98

### Client operating system

Windows

### Server operating system

Ubuntu 20.04 LTS

### Web server

Nginx

### Database engine version

PostgreSQL

### PHP engine version

PHP 8.0

### Nextcloud version

23.0.2

### Updated from an older installed version or fresh install

Updated from an older version

### List of activated apps

```Text
- admin_audit: 1.13.0
  - apporder: 0.14.0
  - bruteforcesettings: 2.3.0
  - calendar: 3.2.0
  - cloud_federation_api: 1.6.0
  - contacts: 4.0.8
  - dav: 1.21.0
  - federatedfilesharing: 1.13.0
  - files: 1.18.0
  - files_external: 1.15.0
  - files_pdfviewer: 2.4.0
  - files_rightclick: 1.2.0
  - files_sharing: 1.15.0
  - files_trashbin: 1.13.0
  - files_versions: 1.16.0
  - files_videoplayer: 1.12.0
  - guests: 2.1.0
  - impersonate: 1.10.0
  - logreader: 2.8.0
  - lookup_server_connector: 1.11.0
  - notifications: 2.11.1
  - notify_push: 0.3.0
  - oauth2: 1.11.0
  - password_policy: 1.13.0
  - photos: 1.5.0
  - previewgenerator: 4.0.0
  - provisioning_api: 1.13.0
  - richdocuments: 5.0.3
  - serverinfo: 1.13.0
  - settings: 1.5.0
  - text: 3.4.0
  - twofactor_backupcodes: 1.12.0
  - twofactor_totp: 6.2.0
  - updatenotification: 1.13.0
  - viewer: 1.7.0
  - workflowengine: 2.5.0
```


### Nextcloud configuration

```Text
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"",
            "192.168.178.10",
            "127.0.1.1"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "23.0.2.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "activity_expire_days": 14,
        "auth.bruteforce.protection.enabled": true,
        "blacklisted_files": [
            ".htaccess",
            "Thumbs.db",
            "thumbs.db"
        ],
        "cron_log": true,
        "default_phone_region": "DE",
        "enable_previews": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\Movie",
            "OC\\Preview\\PDF",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\HEIC",
            "OC\\Preview\\MP4",
            "OC\\Preview\\AVI"
        ],
        "filesystem_check_changes": 0,
        "filelocking.enabled": "true",
        "htaccess.RewriteBase": "\/",
        "integrity.check.disabled": false,
        "knowledgebaseenabled": false,
        "logfile": "\/var\/nc_data\/nextcloud.log",
        "loglevel": 2,
        "logtimezone": "Europe\/Berlin",
        "log_rotate_size": 104857600,
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "overwriteprotocol": "https",
        "preview_max_x": 1920,
        "preview_max_y": 1080,
        "preview_max_scale_factor": 1,
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "quota_include_external_storage": false,
        "share_folder": "\/",
        "skeletondirectory": "",
        "theme": "",
        "trashbin_retention_obligation": "auto, 7",
        "updater.release.channel": "stable",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_smtpport": "465",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "onlyoffice": {
            "jwt_secret": "***REMOVED SENSITIVE VALUE***"",
            "jwt_header": "Authorization",
            "verify_peer_off": true
        },
        "app_install_overwrite": [
            "maps",
            "documentserver_community",
            "onlyoffice",
            "keeweb",
            "flowupload",
            "cookbook",
            "issuetemplate"
        ],
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***"",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "has_rebuilt_cache": true,
        "data-fingerprint": "***REMOVED SENSITIVE VALUE***"",
        "versions_retention_obligation": "auto,3"
    }
}
```


### Web server error log

```Text
192.168.178.54 - - [09/Mar/2022:20:53:59 +0100] "GET /apps/calendar/js/calendar-main.js.map?v=082bf02228876f23fc2e HTTP/2.0" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0"
192.168.178.54 - - [09/Mar/2022:20:53:59 +0100] "GET /apps/files_pdfviewer/js/files_pdfviewer-public.js.map?v=9fb92fbe0bb162b8ebde HTTP/2.0" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0"
192.168.178.54 - - [09/Mar/2022:20:53:59 +0100] "GET /apps/calendar/js/calendar-main.js.map?v=082bf02228876f23fc2e HTTP/2.0" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0"
192.168.178.54 - - [09/Mar/2022:20:53:59 +0100] "GET /apps/calendar/js/calendar-node_modules_moment_locale_de_js.js.map?v=060fd516d6c7c7f54ca9 HTTP/2.0" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0"
```


### Log file

```Text
[no app in context] Warning: Invalid force_event_alarm_type data provided to provideInitialState by calendar

GET /apps/calendar/
from 192.168.178.54 by admin at 2022-03-09T20:41:48+01:00
```


### Browser log

```Text
No OC found index.js:46:12
[DEBUG] @nextcloud/vue: Using version 2.2019c of the timezone database 
Object { app: "@nextcloud/vue" }
ConsoleLogger.js:29:16
@nextcloud/calendar-js app is using version 2.2019c of the timezone database index.esm.js:14165:2
Error: Could not find initial state hide_event_export of calendar
    loadState index.js:24
    e Calendar.vue:221
    u dist.js:1
    _invoke dist.js:1
    S dist.js:1
    XF AppointmentConfigList.vue:127
    o AppointmentConfigList.vue:127
    tx AppointmentConfigList.vue:127
    tx AppointmentConfigList.vue:127
    beforeMount Calendar.vue:207
    VueJS 27
    <anonymous> main.js:76
    <anonymous> main.js:83
    <anonymous> main.js:83
vue.runtime.esm.js:1897:4
    VueJS 4
session heartbeat polling started session-heartbeat.js:101:9
[DEBUG] Files_PDFViewer: Initializing for public page 
Object { app: "Files_PDFViewer", isPublicPage: false, canDownload: true, isSecureViewerAvailable: false }
ConsoleLogger.js:42:18
[ERROR] Files_PDFViewer: But this does not appear to be a public page 
Object { app: "Files_PDFViewer" }
ConsoleLogger.js:54:18
    value ConsoleLogger.js:54
    value ConsoleLogger.js:80
    <anonym> public.js:74
[INFO] calendar: Updated moment locale: de 
Object { app: "calendar" }
ConsoleLogger.js:33:16
OC.requestToken changed lS/BcXI/6lAT+VjUmonSLvoYx5D0S/4xkIXZVqzzXUc=:/R33PhVTnjtY0hXkrsCdaJAggOe8Op0E0+mRPPW0BS4= index.js:321:9
```


### Additional info

checked another installation, same error (Debian 10 and MariaDB 10.6)

--- a/lib/Controller/PublicViewController.php
+++ b/lib/Controller/PublicViewController.php
@@ -145,6 +145,7 @@ class PublicViewController extends Controller {
 		$this->initialStateService->provideInitialState($this->appName, 'default_reminder', $defaultDefaultReminder);
 		$this->initialStateService->provideInitialState($this->appName, 'show_tasks', $defaultShowTasks === 'yes');
 		$this->initialStateService->provideInitialState($this->appName, 'tasks_enabled', false);
+		$this->initialStateService->provideInitialState($this->appName, 'hide_event_export', false);
 
 		return new TemplateResponse($this->appName, 'main', [
 			'share_url' => $this->getShareURL(),

